### PR TITLE
Vuex to pinia

### PR DESCRIPTION
**Pull request by katexhuang**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 ## Project setup
 * Download the [.env](https://start.1password.com/open/i?a=7BICDIKH2ZHQZIH6N3APRMZKLU&v=zu4fcddpxze65mjtzpq6fcadim&i=4qxmcvqnezauxblqwo7hpzzlnu&h=team-swec.1password.ca) and put it in the project folder
 * Run ```make get```
+  * If there is an issue with installing Pinia, try using npm 7 (```nvm use 15```)
```
Comment (chrispellett) [inline README.md:11]: Just want to double check this is expected? Says npm 7 and then 15

File: src/store/asana/worker.ts [hunk 2]
```diff
@@ -13,38 +13,40 @@ export function startWorkers() {
...
   self.setTimeout(() => {
     processActions();
   }, nextTimeout);
 }
 
 async function processAction(): Promise<void> {
-  const state = store.state["asana"];
-  while (state.actions.length > 0) {
-    const action = state.actions[0];
+  const asanaStore = useAsanaStore();
```
Comment (chrispellett) [inline src/store/asana/worker.ts:23]: Probably best to pass this in from the outer function

File: src/store/auth/index.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        "?client_id=1201298517859389" +
+        "&redirect_uri=" + location.protocol + "//" + location.host + "/api" +
+        "&response_type=code" +
+        "&code_challenge_method=S256" +
+        "&code_challenge=" + codeChallenge;
+
+      self.location.href = url;
+    },
+
+    LOGIN_FRON_SESSION(): void {
```
Comment (chrispellett) [inline src/store/auth/index.ts:36]: Spelling

File: src/store/auth/index.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+      asanaStore.$reset();
+      this.user = null;
+    },
+
+    SET_USER(user: User): void {
+      this.user = user;
+      jsonstore.set("user", user);
+    },
+
+    TOKEN_RECIEVED(payload: any): boolean {
```
Comment (chrispellett) [inline src/store/auth/index.ts:60]: Spelling

**Review by chrispellett — APPROVED** [review]
Minor comments, then good to go!

File: README.md (hunk 1 above)
Comment (katexhuang) [reply README.md:11]: Yes, node v15 corresponds with npm v7.

**Pull request merged**